diff -urN mlt-6.2.0.org/src/modules/qt/configure mlt-6.2.0/src/modules/qt/configure
--- mlt-6.2.0.org/src/modules/qt/configure	2016-04-21 04:24:55.000000000 +0200
+++ mlt-6.2.0/src/modules/qt/configure	2016-08-28 11:03:32.132280255 +0200
@@ -83,7 +83,7 @@
 		echo "- Libexif not found, disabling exif features (auto rotate)"
 	fi
 
-	if [ -d "$qt_libdir" -a -d "$qt_includedir" ]
+	if [ 1 -eq 0 -a -d "$qt_libdir" -a -d "$qt_includedir" ]
 	then
 		# test if we have a Qt5 or Qt4
 		if [ -f "$qt_libdir/libQt5Core.so" ] || [ -d "$qt_libdir/QtWidgets.framework" ] || [ -f "$qt_libdir/libQt5Core.a" ]
@@ -144,7 +144,7 @@
 			echo QTLIBS=-L$qt_libdir -lQtCore -lQtGui -lQtXml -lQtSvg -lQtOpenGL >> config.mak
 		fi
 	else
-		pkg-config --exists 'QtGui >= 4'
+		pkg-config --exists 'DISABLE-QtGui >= 4'
 		if [ $? -eq 0 ]
 		then
 			echo "- Qt version 4.x detected"
