diff -dur -x '*~' -x '*.orig' mlt-6.4.1.orig/src/framework/mlt_property.h mlt-6.4.1/src/framework/mlt_property.h
--- mlt-6.4.1.orig/src/framework/mlt_property.h	2016-11-16 07:53:11.000000000 +0100
+++ mlt-6.4.1/src/framework/mlt_property.h	2017-12-10 21:08:17.725929886 +0100
@@ -30,8 +30,10 @@
 #include <sys/param.h>
 #endif
 
-#if defined(__GLIBC__) || defined(__APPLE__) || (__FreeBSD_version >= 900506)
+#if defined(__APPLE__) || (__FreeBSD_version >= 900506)
 #include <xlocale.h>
+#elif defined(__GLIBC__)
+#include <locale.h>
 #else
 typedef char* locale_t;
 #endif
